src/corehost/cli/runtime_config.h

```diff
@@ -18,45 +19,110 @@ enum class roll_fwd_on_no_candidate_fx_option
     major_or_minor
 };
 
+class runtime_config_framework_t
+{
+public:
+    // Uses a "nullable<T>" pattern until we add such a type
+
+    runtime_config_framework_t()
+        : has_fx_ver(false)
+        , has_roll_fwd_on_no_candidate_fx(false)
+        , has_patch_roll_fwd(false)
+        , fx_ver(_X(""))
+        , patch_roll_fwd(false)
+        , roll_fwd_on_no_candidate_fx((roll_fwd_on_no_candidate_fx_option)0)
+        { }
+
+    const pal::string_t* get_fx_ver() const
+    {
+        return (has_fx_ver ? &fx_ver : nullptr);
+    }
+    void set_fx_ver(pal::string_t value)
+    {
+        has_fx_ver = true;
+        fx_ver = value;
+    }
+
+    const bool* get_patch_roll_fwd() const
+    {
+        return (has_patch_roll_fwd ? &patch_roll_fwd : nullptr);
+    }
+    void set_patch_roll_fwd(bool value)
+    {
+        has_patch_roll_fwd = true;
+        patch_roll_fwd = value;
+    }
+
+    const roll_fwd_on_no_candidate_fx_option* get_roll_fwd_on_no_candidate_fx() const
+    {
+        return (has_roll_fwd_on_no_candidate_fx ? &roll_fwd_on_no_candidate_fx : nullptr);
+    }
+    void set_roll_fwd_on_no_candidate_fx(roll_fwd_on_no_candidate_fx_option value)
+    {
+        has_roll_fwd_on_no_candidate_fx = true;
+        roll_fwd_on_no_candidate_fx = value;
+    }
+
+private:
+    bool has_fx_ver;
```

Comment:
Can you just check if `fx_ver` is empty instead of needing to track another variable?

Reply:
I was being consistent with the pattern. So assigning a "" to version would mean it would try to set a "" version during defaulting (which may lead to a desired error later on). If I check for empty, it wouldn't do that.